src/Data/StringVariants/NonEmptyText.hs

```diff
@@ -100,21 +102,22 @@ takeNonEmptyTextEnd :: forall m n. (KnownNat m, KnownNat n, n <= m) => NonEmptyT
 takeNonEmptyTextEnd (NonEmptyText t) =
   if m == n
     then NonEmptyText t
-    else NonEmptyText $ T.takeEnd n t
+    -- when the input is stripped, taking from it is guaranteed to be not empty
+    else NonEmptyText $ T.stripStart $ T.takeEnd n t
   where
     m = fromIntegral $ natVal (Proxy @m)
     n = fromIntegral $ natVal (Proxy @n)
 
 -- | /O(n)/ Splits a 'NonEmptyText' into components of length @chunkSize@. The
--- last element may be shorter than the other chunks, depending on the length
--- of the input.
+-- chunks may be shorter than the chunkSize depending on the length
+-- of the input and spacing. Each chunk is stripped of whitespace.
 chunksOfNonEmptyText ::
   forall chunkSize totalSize.
   (KnownNat chunkSize, KnownNat totalSize) =>
```

Comment:
Do we want a `chunkSize < totalSize` constraint?

Reply:
Yeah, maybe that can catch some typos. It's not necessary for keeping the output correct, though. If a chunk size is larger than the text, it would just widen the string and wrap it in a list.